### PR TITLE
Virtual states, d_a_npc_rival_lod OK, d_a_npc_saltalk almost

### PR DESCRIPTION
**Pull request by robojumper**

### CODE REVIEW

**Comment by robojumper** [comment]
I can't solve the final regswaps of d_a_npc_saltalk (which actually uses virtual states) but I matched d_a_npc_rival_lod instead which gives us a nice regression test for continued work on the base NPC class / virtual weak function order.

File: include/toBeSorted/area_utils.h [hunk 1]
```diff
@@ -1,10 +1,19 @@
 #ifndef AREA_UTILS_H
 #define AREA_UTILS_H
 
+#include "m/m_angle.h"
 #include "m/m_vec.h"
 
-/* 800a9b00 */ bool checkPosInArea(void **AREA, const mVec3_c &pos);
-/* 800a9bd0 */ void *getAreaForIndexInRoom(s32 areaIndex, s32 roomId);
-/* 800a9c50 */ bool *checkPosInArea(s32 areaIndex, s32 roomId, const mVec3_c &pos, void **AREA);
+struct AREA {
```
Comment (elijah-thomas774) [inline include/toBeSorted/area_utils.h:7]: Is this Area the one seen in bzs?
We should probably have proper BZS entries somewhere

Comment (robojumper) [reply include/toBeSorted/area_utils.h:7]: I think so, yes. Let's do that in a separate PR though because I haven't really looked at BZS yet

Comment (elijah-thomas774) [reply include/toBeSorted/area_utils.h:7]: Sounds good

**Review by elijah-thomas774 — APPROVED** [review]

**Pull request merged**